### PR TITLE
[Feature] Storage wrapper update

### PR DESCRIPTION
**Pull request by makylfang**
# Pull Request Template

## Changes

- Added functions to update local and session storage with fallback to in memory
- Updated usages

## Checklist

- [] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/utils/storage.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+  hasMfa?: boolean | undefined
+  pk?: string | undefined
+}
+
+type NFTItem = NFT & {
+  autodetected: boolean
+  chainId: number
+}
+
+function is3PCEnabled() {
```
Comment (mmjee) [inline src/utils/storage.ts:32]: `are3PCEnabled`? What would ‘is 3rd party cookie enabled’  mean?

File: src/utils/storage.ts [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+}
+
+interface IStorage {
+  set(key: string, value: unknown): void
+  get<T>(key: string): T | null
+  delete(key: string): void
+}
+
+class InMemoryStorage implements IStorage {
+  private map = {}
```
Comment (mmjee) [inline src/utils/storage.ts:217]: Simply storing things as properties of an object may cause things like prototype pollution and cause vulnerabilities. I don't know if we store untrusted keys but using a `Map` would be safer here.

**Review by mmjee — APPROVED** [review]
LGTM except the 2 comments

File: src/utils/storage.ts (hunk 2 above)
Comment (makylfang) [reply src/utils/storage.ts]: @mmjee updated

File: src/utils/storage.ts (hunk 1 above)
Comment (makylfang) [reply src/utils/storage.ts]: @mmjee updated

**Review by mmjee — APPROVED** [review]
👍🏼

**Pull request merged**